### PR TITLE
[#356] tqdm and logging in embedding pipeline

### LINKED ISSUE #356: Tqdm in embedding
Add logging and tqdm (progress bars) in embedding pipeline

### PR DESCRIPTION
**Pull request by Michal-Kolomanski**
closes #356

Embedding looks like this right now:
Screenshot from 2022-04-25 11-00-13

All logs may be disabled using (`verbose=False`)

### CODE REVIEW

File: recommender/engines/autoencoders/inference/embedding_component.py [hunk 1]
```diff
@@ -18,18 +18,43 @@ class EmbeddingComponent:
...
+        if verbose:
+            logger.info("Start embedding...")
+            logger.info("Start converting one_hot_tensors into dense_tensors...")
+
+        self.user_embedder(
+            User.objects,
+            use_cache=use_cache,
+            save_cache=save_cache,
+            version="Users",
+            verbose=True,
```
Comment (wujuu) [inline recommender/engines/autoencoders/inference/embedding_component.py:37]: shouldn't we pass verbose from the function call here?

File: recommender/engines/autoencoders/inference/embedding_component.py [hunk 2]
```diff
@@ -18,18 +18,43 @@ class EmbeddingComponent:
...
+        )
         self.services_embedder(
-            Service.objects, use_cache=use_cache, save_cache=save_cache
+            Service.objects,
+            use_cache=use_cache,
+            save_cache=save_cache,
+            version="Services",
+            verbose=True,
```
Comment (wujuu) [inline recommender/engines/autoencoders/inference/embedding_component.py:44]: same as higher

File: recommender/engines/autoencoders/inference/embedding_component.py [hunk 3]
```diff
@@ -18,18 +18,43 @@ class EmbeddingComponent:
...
-        self.normalizer(User.objects, save_cache=save_cache)
-        self.normalizer(Service.objects, save_cache=save_cache)
-        logger.info("Finished embedding!")
+
+        if verbose:
+            logger.info("Finished converting one_hot_tensors into dense_tensors!")
+            logger.info("Start normalizing dense_tensors...")
+
+        self.normalizer(
+            User.objects, save_cache=save_cache, version="Users", verbose=True
```
Comment (wujuu) [inline recommender/engines/autoencoders/inference/embedding_component.py:52]: same as higher

File: recommender/engines/autoencoders/inference/embedding_component.py [hunk 4]
```diff
@@ -18,18 +18,43 @@ class EmbeddingComponent:
...
+        )
+        self.normalizer(
+            Service.objects, save_cache=save_cache, version="Services", verbose=True
```
Comment (wujuu) [inline recommender/engines/autoencoders/inference/embedding_component.py:55]: same as higher

File: recommender/engines/autoencoders/ml_components/embedder.py [hunk 5]
```diff
@@ -72,16 +79,24 @@ def __call__(
...
-        one_hot_tensors = [obj.one_hot_tensor for obj in objects]
+        if verbose:
+            logger.info("Collecting one_hot_tensors from %s objects", version)
+        one_hot_tensors = [
+            obj.one_hot_tensor
+            for obj in tqdm(objects, desc=version, disable=not verbose)
+        ]
         one_hot_tensors_batch = torch.Tensor(one_hot_tensors)
 
         dense_tensors_batch = self.network(one_hot_tensors_batch)
```
Comment (wujuu) [inline recommender/engines/autoencoders/ml_components/embedder.py:91]: add logging of the network feed-forward

File: recommender/commands/train.py [hunk 6]
```diff
@@ -44,7 +44,7 @@ def _rl_v2():
 def _all():
     """Run all training pipelines"""
     AEPipeline(AUTOENCODERS_PIPELINE_CONFIG)()
-    EmbeddingComponent()()
+    EmbeddingComponent()(verbose=True)
```
Comment (wujuu) [inline recommender/commands/train.py:47]: we shloud delegate choice of being verbose or not to the user (developer) calling our cli - e.g add verbose option to click

**Review by wujuu — APPROVED** [review]

File: recommender/commands/train.py (hunk 6 above)
Comment (Michal-Kolomanski) [reply recommender/commands/train.py:47]: I agree but also I think that it should be the scope for a new ticket - make it easy to enable or disable being verbose for every command.

Comment (Michal-Kolomanski) [reply recommender/commands/train.py:47]: I made a ticket https://github.com/cyfronet-fid/recommender-system/issues/360

**Review by wujuu — APPROVED** [review]

**Pull request merged**